Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/non_rigid_mesh_registration.h

```diff
@@ -0,0 +1,799 @@
+// Copyright (c) 2024 GeometryFactory (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+//
+// $URL$
+// $Id$
+// SPDX-License-Identifier: GPL-3.0-or-later OR LicenseRef-Commercial
+//
+// Author(s)     : Roberto Dyke, Sven Oesau
+//
+#ifndef CGAL_POLYGON_MESH_PROCESSING_REGISTER_MESH_H
+#define CGAL_POLYGON_MESH_PROCESSING_REGISTER_MESH_H
+
+#include <CGAL/license/Polygon_mesh_processing/miscellaneous.h>
+
+#include <CGAL/config.h>
+
+//#ifdef CGAL_EIGEN3_ENABLED
```

Comment:
In case we keep Eigen as implementation detail the doxygen documentation of the functions needs a `@note`  about the dependency.

Reply:
Now there is a `@note`.   There is the other problem that the function is now not defined when Eigen is not enabled.  Is that what we do in other packages.    I would expect a compilation error message that tells me that the absence of Eigen is the problem, and not that the function is not found.